.github/workflows/rust.yml

```
@@ -0,0 +1,145 @@
+name: Rust
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    name: Build Napture (Ubuntu)
+    runs-on: ubuntu-24.04
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install Dependencies
+      run: |
+        echo "Installing dependencies..."
+        sudo apt-get update
+        sudo apt-get install -y libglib2.0-dev libcairo2-dev libgraphene-1.0-dev libgtk-4-dev libadwaita-1-dev 
+        sudo add-apt-repository universe
+        sudo apt install libfuse2
+        echo "Dependencies installed."
+
+    - name: Build Napture
+      run: |
+        echo "Building Napture..."
+        cd napture
+        export PKG_CONFIG_PATH=/usr/lib/pkgconfig
+        cargo build --release
+        echo "Napture built successfully."
+    - name: Install AppImageTool
+      run: |
+        echo "Starting artifact upload phase..."
+        echo "Installing AppImageTool..."
+        wget https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage
+        chmod +x appimagetool-x86_64.AppImage
+        echo "AppImageTool installed."
+    - name: Create Napture AppImage
+      run: |
+        echo "Creating Napture AppImage..."
+        cd napture
+        mkdir -p AppDir/usr/bin
+        cp target/release/webx AppDir/usr/bin/
+        cp io.github.face_hh.Napture.desktop AppDir/
+        cp io.github.face_hh.Napture.svg AppDir/
+        ARCH=x86_64
+         ../appimagetool-x86_64.AppImage AppDir
+        mv Napture-x86_64.AppImage ../Bussin.Napture-x86_64.AppImage
+        echo "AppImage created successfully."
+    - name: Rename AppImage
+      run: |
+        echo "Renaming AppImage..."
+        mv Bussin.Napture-x86_64.AppImage napture-v${{ github.sha }}.AppImage
+        echo "AppImage renamed successfully."
```

It'd be better to use linuxdeploy, as that also packs the dependencies and everything
```suggestion
    - name: Install linuxdeploy
      run: |
        echo "Starting artifact upload phase..."
        echo "Installing linuxdeploy..."
        wget https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage
        chmod +x linuxdeploy-x86_64.AppImage
        echo "linuxdeploy installed."
    - name: Create Napture AppImage
      run: |
        echo "Creating Napture AppImage..."
        cd napture
        mv target/release/webx target/release/napture
        ARCH=x86_64
         ../linuxdeploy-x86_64.AppImage --appdir AppDir -e target/release/napture -d io.github.face_hh.Napture.desktop -i io.github.face_hh.Napture.svg
        mv Napture-x86_64.AppImage ../Bussin.Napture-x86_64.AppImage
        echo "AppImage created successfully."
    - name: Rename AppImage
      run: |
        echo "Renaming AppImage..."
        mv Bussin.Napture-x86_64.AppImage napture-v${{ github.sha }}.AppImage
        echo "AppImage renamed successfully."
```